### PR TITLE
Add label to Checkbox widget and fix examples

### LINKED ISSUE #635: checkbox widget should have a label
I don't think it's meaningful to have a checkbox without text, although we can expose a `RawCheckbox` if we wish. 

The checkbox should have the same general layout properties as the radio widget, and clicking on the label should toggle the check.

### PR DESCRIPTION
**Pull request by futurepaul**
closes #635 

Forgoing `RawCheckbox` for simplicity, but someone can feel free to add it if they need it!

### CODE REVIEW

File: druid/src/widget/checkbox.rs [hunk 1]
```diff
@@ -69,16 +73,29 @@ impl Widget<bool> for Checkbox {
...
-        _layout_ctx: &mut LayoutCtx,
+        layout_ctx: &mut LayoutCtx,
         bc: &BoxConstraints,
-        _data: &bool,
+        data: &bool,
         env: &Env,
     ) -> Size {
         bc.debug_check("Checkbox");
 
+        // bc.constrain(Size::new(
```
Comment (cmyr) [inline druid/src/widget/checkbox.rs]: ✂️

File: druid/src/widget/checkbox.rs [hunk 2]
```diff
@@ -69,16 +73,29 @@ impl Widget<bool> for Checkbox {
...
+        //     env.get(theme::BASIC_WIDGET_HEIGHT),
+        //     env.get(theme::BASIC_WIDGET_HEIGHT),
+        // ))
+
+        let label_size = self.child_label.layout(layout_ctx, &bc, data, env);
+        let padding = 8.0;
```
Comment (cmyr) [inline druid/src/widget/checkbox.rs]: it feels like this should maybe be a constant somewhere, and at least shared between the radio button and the checkbox? That can be a separate PR, though.

`CONTROL_WIDGET_HORIZ_PADDING`?

Comment (futurepaul) [reply druid/src/widget/checkbox.rs]: Yeah I'll do this along with fixing the padding of radio.

File: druid/src/widget/checkbox.rs (hunk 1 above)
Comment (futurepaul) [reply druid/src/widget/checkbox.rs]: Oops thank you!

**Pull request merged**